tests/test_date.py

```diff
@@ -2,90 +2,158 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
 
+from types import MethodType
+
+from collections import OrderedDict
+
 import unittest
 from datetime import datetime, timedelta
 
 from dateutil.relativedelta import relativedelta
 from mock import Mock, patch
+
 from nose_parameterized import parameterized, param
 
+from dateparser.languages import LanguageDataLoader
+
 from dateparser import date
 from tests import BaseTestCase
 
 
-class DateRangeTest(unittest.TestCase):
+class DateRangeTest(BaseTestCase):
+    def setUp(self):
+        super(DateRangeTest, self).setUp()
 
-    def test_should_render_10_days_range(self):
-        begin = datetime(2014, 6, 15)
-        end = datetime(2014, 6, 25)
-        dates = list(date.date_range(begin, end))
-        self.assertEquals(10, len(dates))
-        self.assertEquals(begin, dates[0])
-        self.assertEquals(end - timedelta(days=1), dates[-1])
+    @parameterized.expand([
+        param(begin=datetime(2014, 6, 15), end=datetime(2014, 6, 25), expected_length=10)
+    ])
+    def test_date_range(self, begin, end, expected_length):
+        self.given_period_start(begin)
+        self.given_period_end(end)
+        self.when_date_range_generated()
+        self.then_range_length_is(expected_length)
+        self.then_all_dates_from_beginning_to_end_are_present_in_range()
+        self.then_range_is_in_ascending_order()
 
-    def test_should_one_date_for_each_month(self):
-        fixtures = [
-            (datetime(2014, 4, 15), datetime(2014, 6, 25),
-             [(2014, 4), (2014, 5), (2014, 6)]),
+    @parameterized.expand([
+        param(begin=datetime(2014, 4, 15),
+              end=datetime(2014, 6, 25),
+              expected_months=[(2014, 4), (2014, 5), (2014, 6)]),
+        param(begin=datetime(2014, 4, 25),
+              end=datetime(2014, 5, 5),
+              expected_months=[(2014, 4), (2014, 5)]),
+        param(begin=datetime(2014, 4, 5),
+              end=datetime(2014, 4, 25),
+              expected_months=[(2014, 4)]),
+        param(begin=datetime(2014, 4, 25),
+              end=datetime(2014, 6, 5),
+              expected_months=[(2014, 4), (2014, 5), (2014, 6)]),
+    ])
+    def test_one_date_for_each_month(self, begin, end, expected_months):
+        self.given_period_start(begin)
+        self.given_period_end(end)
+        self.given_period_size(months=1)
+        self.when_month_range_generated()
+        self.then_month_expected(expected_months)
 
-            (datetime(2014, 4, 25), datetime(2014, 5, 5),
-             [(2014, 4), (2014, 5)]),
+    @parameterized.expand([
+        param(begin=datetime(2014, 6, 15), end=datetime(2014, 6, 25))
+    ])
+    def test_should_reject_easily_mistaken_dateutil_arguments(self, begin, end):
```

Comment:
Yup, this even [already implemented](https://github.com/scrapinghub/dateparser/pull/56/files?diff=unified#diff-1899ed0097831b86662064a1e60ee782R196).